pkg/flare/archive.go

```diff
@@ -117,11 +123,39 @@ func CreatePerformanceProfile(pprofURL string, profileDuration int) (*Profile, e
 		return nil, err
 	}
 
-	return &Profile{
+	p := &Profile{
 		FirstHeapProfile:  firstHeapProf,
 		SecondHeapProfile: secondHeapProf,
 		CPUProfile:        cpuProf,
-	}, nil
+	}
+
+	if k := "apm_config.enabled"; config.Datadog.IsSet(k) && !config.Datadog.GetBool(k) {
```

Comment:
Yes, because when the value isn't set, the return from `GetBool` is `false`, which isn't true because if the option isn't set APM is enabled by default. Hope that makes sense.